server/controllers/userControllers.js

```diff
@@ -15,39 +15,63 @@ class userControllers {
       phoneNumber,
       username,
     } = req.body;
-    const newUser = [
-      {
-        id: Users.length + 1,
-        firstname,
-        lastname,
-        email,
-        password,
-        phoneNumber,
-        username,
-        isAdmin: false,
-      },
-    ];
+    const newUser = {
+      firstname,
+      lastname,
+      email,
+      password,
+      phoneNumber,
+      username,
+      isAdmin: false,
+    };
+
     const hashedPassword = hashPassword(password);
     const userToken = signJwt(newUser);
     newUser.password = hashedPassword;
-    if (email === newUser.email) {
+    const userEmail = newUser.email;
+    if (email === userEmail) {
       Users.push(newUser);
       return res.status(201).json({
         status: res.statusCode,
         message: 'user Created successfull',
         data: {
           userToken,
+          id: Users.length + 1,
           firstname,
           lastname,
           email,
           phoneNumber,
           username,
+          isAdmin: false,
         },
       });
     }
     return res.status(400).json({
       status: res.statusCode,
-      message: 'email doesn\'t match',
+      message: "email doesn't match",
+    });
+  }
+
+  //   User signin Controllers
+  static userSignin(req, res) {
+    const { email, password } = req.body;
+    const userwithEmail = Users.find((user) => user.email === email);
+    const userEmail = userwithEmail.email;
+    const findPassword = userwithEmail.password;
+    const compare = compareHash(password, findPassword);
+    const userToken = signJwt(email);
+    if (userEmail === email && compare === true) {
```

Comment:
@mic7x4 why this condition?

Reply:
I wanna check if the email given and the password user typed is equal to the hashed password